sdks/python/apache_beam/coders/coders.py

```diff
@@ -1383,22 +1385,50 @@ def from_runner_api_parameter(payload, components, context):
         write_state_threshold=int(payload))
 
 
-class RunnerAPICoderHolder(Coder):
+class CoderElementType(typehints.TypeConstraint):
+  """An element type that just holds a coder proto."""
+  def __init__(self, coder_proto, context):
+    self.coder_proto = coder_proto
```

Comment:
Logically the ElementType should hold the ExternalCoder, and the ExternalCoder should hold the proto. (This will also help some issues where coders are requested manually and then not respected.)